src/blob_gc_job.cc

```diff
@@ -581,5 +581,73 @@ bool BlobGCJob::IsShutingDown() {
   return (shuting_down_ && shuting_down_->load(std::memory_order_acquire));
 }
 
+Status BlobGCJob::DigHole() {
+  Status s;
+  const auto& inputs = blob_gc_->sampled_inputs();
+  assert(!inputs.empty());
+  for (std::size_t i = 0; i < inputs.size(); ++i) {
+    std::unique_ptr<PosixRandomRWFile> file;
+    // TODO(@DorianZheng) set read ahead size
+    s = OpenBlobFile(inputs[i]->file_number(), 0, db_options_, env_options_,
+                     env_, &file);
+    if (!s.ok()) {
+      break;
+    }
+
+    auto gc_iter = new BlobFileIterator(
+        std::move(file), inputs[i]->file_number(), inputs[i]->file_size(),
+        blob_gc_->titan_cf_options());
+
+    std::string last_key;
+    bool last_key_valid = false;
+    uint64_t last_valid_tail = 0;
+    uint64_t cur_valid_head = 0;
+    gc_iter->SeekToFirst();
+    assert(gc_iter->Valid());
```

Comment:
SeekToFirst() may fail, we'd better check the status rather than assert it.

Reply:
for-loop will check the status.